plugins/in_winevtlog/winevtlog.c

```diff
@@ -0,0 +1,786 @@
+    if (hContext) {
+        EvtClose(hContext);
+    }
+
+    if (hMetadata) {
+        EvtClose(hMetadata);
+    }
+
+    return wMessage;
+}
+
+int get_string_inserts(EVT_HANDLE handle, PEVT_VARIANT *values, UINT *prop_count, unsigned int *string_inserts_size)
+{
+    PEVT_VARIANT pValues;
+    DWORD dwBufferSize = 0;
+    DWORD dwBufferSizeUsed = 0;
+    DWORD dwPropCount = 0;
+    BOOL succeeded = FLB_TRUE;
+
+    EVT_HANDLE hContext = EvtCreateRenderContext(0, NULL, EvtRenderContextUser);
+    if (hContext == NULL) {
+        flb_error("Failed to create renderContext");
+        succeeded = FLB_FALSE;
+        goto cleanup;
+    }
+
+    // Get the size of the buffer
+    EvtRender(hContext, handle, EvtRenderEventValues, 0, NULL, &dwBufferSize, &dwPropCount);
+    pValues = (PEVT_VARIANT)flb_malloc(dwBufferSize);
+
+    succeeded = EvtRender(hContext,
+                          handle,
+                          EvtRenderContextValues,
+                          dwBufferSize,
+                          pValues,
+                          &dwBufferSizeUsed,
+                          &dwPropCount);
+
+    if (!succeeded) {
+        flb_error("Failed to get string inserts with %d\n", GetLastError());
+        goto cleanup;
+    }
+
+    *prop_count = dwPropCount;
+    *values = pValues;
+    *string_inserts_size = dwBufferSize;
+
+cleanup:
+
+    if (hContext != NULL)
+        EvtClose(hContext);
+
+    return succeeded;
+}
+
+static int winevtlog_next(struct winevtlog_channel *ch, int hit_threshold)
+{
+    EVT_HANDLE hEvents[SUBSCRIBE_ARRAY_SIZE];
+    DWORD count = 0;
+    DWORD status = ERROR_SUCCESS;
+
+    /* If subscription handle is NULL, it should return false. */
+    if (!ch->subscription) {
+        flb_error("Invalid subscription is passed");
+        return FLB_FALSE;
+    }
+
+    if (hit_threshold) {
+        return FLB_FALSE;
+    }
+
+    if (!EvtNext(ch->subscription, SUBSCRIBE_ARRAY_SIZE, hEvents, INFINITE, 0, &count)) {
+        status = GetLastError();
+        if (ERROR_CANCELLED == status) {
+            return FLB_FALSE;
+        }
+        if (ERROR_NO_MORE_ITEMS != status) {
+            return FLB_FALSE;
+        }
+    }
+
+    if (status == ERROR_SUCCESS) {
+        ch->count = count;
+        for (int i = 0; i < count; i++) {
+            ch->events[i] = hEvents[i];
+            EvtUpdateBookmark(ch->bookmark, ch->events[i]);
+        }
+
+        return FLB_TRUE;
+    }
+
+    return FLB_FALSE;
+}
+
+/*
+ * Read from an open Windows Event Log channel.
+ */
+int winevtlog_read(struct winevtlog_channel *ch, msgpack_packer *mp_pck, struct winevtlog_config *ctx,
+                   unsigned int *read)
+{
+    DWORD status = ERROR_SUCCESS;
+    PWSTR wSystem = NULL;
+    unsigned int system_size = 0, message_size = 0, string_inserts_size = 0;
+    int hit_threshold = FLB_FALSE;
+    unsigned int read_size = 0;
+    PWSTR wMessage = NULL;
+    PEVT_VARIANT pSystem = NULL;
+    PEVT_VARIANT pStringInserts = NULL;
```

Comment:
Could we not mix camel case and underscores at least in the same file? I'm against camel case and any windows convention because it makes it harder to get acclimated to the source code when switching between files.

Reply:
Fixed in https://github.com/fluent/fluent-bit/pull/4179/commits/aae73fa0a70fa377af361f379601de7a0fc3a1e7.